pkg/operator/ceph/file/mds/config.go

```diff
@@ -74,6 +74,9 @@ func (c *Cluster) setDefaultFlagsMonConfigStore(mdsID string) error {
 	if !c.fs.Spec.MetadataServer.Resources.Limits.Memory().IsZero() {
 		mdsCacheMemoryLimit := float64(c.fs.Spec.MetadataServer.Resources.Limits.Memory().Value()) * mdsCacheMemoryLimitFactor
 		configOptions["mds_cache_memory_limit"] = strconv.Itoa(int(mdsCacheMemoryLimit))
+	} else if !c.fs.Spec.MetadataServer.Resources.Requests.Memory().IsZero() {
```

Comment:
How about if we put this check for requests first, and then the check for limits second? (switch the order of lines 74 and 77). I think this will match the expectations from dissussion in #8143.

Reply:
done. 
```
limits:
       cpu: "500m"
       memory: "3Gi"
     requests:
       cpu: "400m"
       memory: "2Gi"
```
`"mds_cache_memory_limit": "1073741824",`